jbake-core/src/main/java/org/jbake/parser/MarkupEngine.java

```diff
@@ -149,50 +153,55 @@ public void processBody(final ParserContext context) {
         return content;
     }
 
+
     /**
      * Checks if the file has a meta-data header.
      *
      * @param contents Contents of file
      * @return true if header exists, false if not
      */
     private boolean hasHeader(Configuration config, List<String> contents) {
-        boolean headerValid = false;
         boolean headerSeparatorFound = false;
         boolean statusFound = false;
         boolean typeFound = false;
 
-        List<String> header = new ArrayList<String>();
+        List<String> headerLines = new ArrayList<String>();
+
+
+        //for (String line : contents) {
+        for (int i = 0; i < contents.size() && i < MAX_HEADER_LINES; i++) {
```

Comment:
I have some files which are thousands lines long. And JBake scans them 3 times. I think we can optimize a bit and apply some configurable limit.